fuel-core-interfaces/src/txpool.rs

```diff
@@ -26,61 +26,66 @@ pub trait TxPoolDb:
     }
 }
 
-/// Subscriber interface that receive inserted/removed events from txpool.
-/// Usually needed for libp2p broadcasting and can be used to notify users of new tx.
-#[async_trait]
-pub trait Subscriber: Send + Sync {
-    async fn inserted(&self, tx: Arc<Transaction>);
-
-    async fn inserted_on_block_revert(&self, tx: Arc<Transaction>);
-
-    async fn removed(&self, tx: Arc<Transaction>, error: &Error);
-}
-
-#[async_trait]
-pub trait TxPool: Send + Sync {
+#[derive(Debug)]
+pub enum TxPoolMpsc {
+    /// Return all sorted transactions that are includable in next block.
+    /// This is going to be heavy operation, use it only when needed.
+    Includable {
+        response: oneshot::Sender<Vec<Arc<Transaction>>>,
+    },
     /// import list of transaction into txpool. All needed parents need to be known
     /// and parent->child order should be enforced in Vec, we will not do that check inside
     /// txpool and will just drop child and include only parent. Additional restrain is that
     /// child gas_price needs to be lower then parent gas_price. Transaction can be received
     /// from p2p **RespondTransactions** or from userland. Because of userland we are returning
     /// error for every insert for better user experience.
-    async fn insert(&self, tx: Vec<Arc<Transaction>>)
-        -> Vec<anyhow::Result<Vec<Arc<Transaction>>>>;
-
+    Insert {
+        txs: Vec<Arc<Transaction>>,
+        response: oneshot::Sender<Vec<anyhow::Result<Vec<Arc<Transaction>>>>>,
+    },
     /// find all tx by their hash
-    async fn find(&self, hashes: &[TxId]) -> Vec<Option<TxInfo>>;
-
+    Find {
+        ids: Vec<TxId>,
+        response: oneshot::Sender<Vec<Option<TxInfo>>>,
+    },
     /// find one tx by its hash
-    async fn find_one(&self, hash: &TxId) -> Option<TxInfo>;
-
+    FindOne {
+        id: TxId,
+        response: oneshot::Sender<Option<TxInfo>>,
+    },
     /// find all dependent tx and return them with requsted dependencies in one list sorted by Price.
-    async fn find_dependent(&self, hashes: &[TxId]) -> Vec<Arc<Transaction>>;
-
+    FindDependent {
+        ids: Vec<TxId>,
+        response: oneshot::Sender<Vec<Arc<Transaction>>>,
+    },
+    /// remove transaction from pool needed on user demand. Low priority
+    Remove { ids: Vec<TxId> },
```

Comment:
Shouldn't we separate debug/helpers from the core functionality?

Reply:
`Remove` is under core functionality, as having the ability to remove transactions from txpool. blacklisting transaction from pool is something that we can think about in future.